src/tests/unittests/asn1.c

```diff
@@ -47,12 +47,91 @@ static void torture_large_oid(void **state)
 	assert_int_equal(rv, SC_ERROR_NOT_SUPPORTED);
 }
 
+static void torture_integer(void **state)
+{
+	/* Without the Tag and Length {0x02, 0x01} */
+	u8 zero[] = {0x00};
+	u8 one[] = {0x01};
+	u8 minus_one[] = {0xFF};
+	u8 max2[] = {0x7F, 0xFF};
+	u8 min2[] = {0x80, 0x00};
+	u8 max4[] = {0x7F, 0xFF, 0xFF, 0xFF};
+	u8 min4[] = {0x80, 0x00, 0x00, 0x00};
+	u8 over[] = {0x7F, 0xFF, 0xFF, 0xFF, 0xFF};
+	int value;
+	int rv = 0;
+
+	rv = sc_asn1_decode_integer(zero, sizeof(zero), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, 0);
+
+	rv = sc_asn1_decode_integer(one, sizeof(one), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, 1);
+
+	rv = sc_asn1_decode_integer(minus_one, sizeof(minus_one), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, (int)-1);
+
+	rv = sc_asn1_decode_integer(max2, sizeof(max2), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, 32767);
+
+	rv = sc_asn1_decode_integer(min2, sizeof(min2), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, -32768);
+
+	if (sizeof(int*) == 8) {
+		/* For 64 bit builds */
+		rv = sc_asn1_decode_integer(max4, sizeof(max4), &value);
+		assert_int_equal(rv, SC_SUCCESS);
+		assert_int_equal(value, 2147483647);
+
+		rv = sc_asn1_decode_integer(min4, sizeof(min4), &value);
+		assert_int_equal(rv, SC_SUCCESS);
+		assert_int_equal(value, -2147483648);
+
+		rv = sc_asn1_decode_integer(over, sizeof(over), &value);
+		assert_int_equal(rv, SC_ERROR_INVALID_ASN1_OBJECT);
```

Comment:
there really isn't a function that can handle integers bigger than INT_MAX? those methods won't be able to decode an ECDSA signature made with P-160, let alone anything actually secure... (or serial number in a typical X.509 certificate, or an RSA public key, etc.)

Reply:
also, it's it more of a overflow rather than invalid asn1 object per-se?

Reply:
If I see right, there are arbitrary large integers handled as octet strings. Other primitives are usually not handled in this asn1 library and go through the OpenSSL.
Good point about the error. I changed that to `NOT_SUPPORTED` to better reflect reality.

Reply:
> If I see right, there are arbitrary large integers handled as octet strings.

the problem with this is that make the implementation accept malformed ones (like ones that use too many bytes to encode a value)

Reply:
This is the  problem that the `asn1_decode_integer()` is used also from other places for parsing the size in less strict inputs. Enforcing the format in dbba826f already broke the functional tests so I will try to modify the code to be less strict about these kinds of errors if asked through argument.

Reply:
both DER and BER encodings require that there are no padding bytes in INTEGER encoding, so if that's too strict, I'd say that those specific usages should use a less strict parser rather than making the default parser non-compliant